sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/BaseStreamingSink.java

```diff
@@ -0,0 +1,7 @@
+package org.apache.spark.sql.execution.streaming;
+
+/**
+ * The shared interface between V1 and V2 streaming sinks.
+ */
+public interface BaseStreamingSink {
```

Comment:
do we have to have this? it may pollute the public interface.

Reply:
What kind of pollution are you concerned about? Maybe we can take steps to minimize it.

The migration from our current internal API will be much harder if we don't have a common interface. We'd need to duplicate StreamExecution and keep both versions around for a reasonably long period, risking confusion and accidental dropping of bugfixes that should've applied to both.

Reply:
Let make it obvious in the docs that this is designed for temporary compatibility and will be removed in future versions.